### PR TITLE
adds new collection type code for new setting

### LINKED ISSUE #533: Create a new option for Collections which enables configurable facets
### Descriptive summary

When creating a new collection type, there are multiple options that you can enable in which "append" functionality to the collection type. There needs to be a choice in the settings tab after you create the new collection type that enables that collection type to have configurable facets. 

The workflow would be as follows:
- [x] Go to collection type tab
- [x] Create new collection type
- [x] Immediately after creation, go to settings tab
- [x] Click the check box to enable configurable faceting
- [x] Save the changes

By following these steps, when you create a collection of the previously created work type, it should allow you to configure the facets for that collection. This includes being able to order the facets, change the label, and enable/disable certain facets. This data will need to be tracked per collection and should be configurable per collection.

### PR DESCRIPTION
**Pull request by straleyb**
This adds the base code for creation of collection types with the new option of configurable facets. 

Fixes #533

### CODE REVIEW

**Comment by straleyb** [comment]
collection_type

File: app/controllers/admin/scholars_archive/collection_types_controller.rb [hunk 1]
```diff
@@ -0,0 +1,17 @@
+# frozen_string_literal: true
+
+module Admin
+  module ScholarsArchive
```
Comment (CGillen) [inline app/controllers/admin/scholars_archive/collection_types_controller.rb]: OregonDigital

File: app/controllers/admin/scholars_archive/collection_types_controller.rb [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+    # Extension of Hyrax CollectionTypesController to add facet_configurable to strong params
+    class CollectionTypesController < Hyrax::Admin::CollectionTypesController
+      self.form_class = ScholarsArchive::Forms::Admin::CollectionTypeForm
```
Comment (CGillen) [inline app/controllers/admin/scholars_archive/collection_types_controller.rb]: OregonDigital::Forms::Admin::CollectionTypeForm

File: app/forms/scholars_archive/forms/admin/collection_type_form.rb [hunk 3]
```diff
@@ -0,0 +1,12 @@
+# frozen_string_literal: true
+
+module ScholarsArchive
```
Comment (CGillen) [inline app/forms/scholars_archive/forms/admin/collection_type_form.rb]: OregonDigital

File: config/routes.rb [hunk 4]
```diff
@@ -1,6 +1,9 @@
 # frozen_string_literal:true
 
 Rails.application.routes.draw do
+  namespace :admin do 
+    resources :collection_types, except: :show, controller: 'scholars_archive/collection_types'
```
Comment (CGillen) [inline config/routes.rb]: oregon_digital/collection_types

File: config/routes.rb [hunk 5]
```diff
@@ -44,6 +47,6 @@
   end
 
   resources :oembeds, controller: 'oregon_digital/oembeds', only: %i[index edit]
-
+
```
Comment (CGillen) [inline config/routes.rb]: Lint

File: spec/views/hyrax/admin/collection_types/_form_settings.html.erb_spec.rb [hunk 6]
```diff
@@ -0,0 +1,103 @@
...
+  INPUT_IDS = %w[
+    collection_type_nestable
+    collection_type_brandable
+    collection_type_discoverable
+    collection_type_sharable
+    collection_type_share_applies_to_new_works
+    collection_type_allow_multiple_membership
+  ].freeze
+
+  let(:collection_type_form) { ScholarsArchive::Forms::Admin::CollectionTypeForm.new }
```
Comment (CGillen) [inline spec/views/hyrax/admin/collection_types/_form_settings.html.erb_spec.rb]: OregonDigital::Forms::Admin::CollectionTypeForm.new

File: spec/views/hyrax/admin/collection_types/_form_settings.html.erb_spec.rb [hunk 7]
```diff
@@ -0,0 +1,103 @@
...
+
+  let(:form) do
+    view.simple_form_for(collection_type, url: '/update') do |fs_form|
+      return fs_form
+    end
+  end
+
+  before do
+    collection_type_form.collection_type = collection_type
+    # allow(view).to receive(:f).and_return(form)
```
Comment (CGillen) [inline spec/views/hyrax/admin/collection_types/_form_settings.html.erb_spec.rb]: Lint

File: spec/views/hyrax/admin/collection_types/_form_settings.html.erb_spec.rb [hunk 8]
```diff
@@ -0,0 +1,103 @@
+# frozen_string_literal: true
+
+RSpec.describe 'hyrax/admin/collection_types/_form_settings.html.erb', type: :view do
+  # TODO: add fields as they become available:
+  # collection_type_assigns_workflow
+  # collection_type_require_membership
+  # collection_type_assigns_visibility
+
+  INPUT_IDS = %w[
+    collection_type_nestable
```
Comment (CGillen) [inline spec/views/hyrax/admin/collection_types/_form_settings.html.erb_spec.rb]: Does this need to add `collection_type_facet_configurable`?

File: spec/views/hyrax/admin/collection_types/_form_settings.html.erb_spec.rb [hunk 9]
```diff
@@ -0,0 +1,103 @@
...
+        end
+      end
+    end
+
+    context 'when collection_type.collections? is true' do
+      before do
+        collection_type_form.collection_type = collection_type
+        allow(collection_type).to receive(:collections?).and_return(true)
+        assign(:form, collection_type_form)
+        # allow(view).to receive(:f).and_return(form)
```
Comment (CGillen) [inline spec/views/hyrax/admin/collection_types/_form_settings.html.erb_spec.rb]: Lint

**Pull request merged**